src/domUpdates.js

```diff
@@ -1,69 +1,182 @@
+          return room.number === booking.roomNumber
+        });
+        cardsSection.innerHTML += `
+        <div class="card">
         <p>Date: ${booking.date}</p>
         <p>Room Number: ${booking.roomNumber}<p>
         <p>Room Type: ${room.roomType}</p>
         <p>Cost: $${room.costPerNight}</p>
-      </div>
+        </div>
+        `;
+      });
+    }
+  },
+
+  populateAvailableRooms() {
+    availableRoomsSection.innerHTML = '';
+    hotel.getAvailableRooms(dateInput.value);
+    domUpdates.checkFilters();
+    if (!hotel.availableRooms.length) {
+      availableRoomsSection.innerHTML =  `
+        <p>Unfortunately, all of our rooms are booked on this day. Please try another date or room type!</p>
       `;
-    });
+    } else {
+      hotel.availableRooms.forEach(room => {
+        const bidet = room.bidet ? 'yes' : 'no';
+        availableRoomsSection.innerHTML += `
+        <div class="card">
+        <p>Room Number: ${room.number}</p>
+        <p>Cost: $${room.costPerNight}<p>
+        <p>Room Type: ${room.roomType}</p>
+        <p>Number of Beds: ${room.numBeds}</p>
+        <p>Bed Size: ${room.bedSize}</p>
+        <p>Bidet: ${bidet}<p>
+        <button class="book-button" id="${room.number}">Book Room</button>
+        </div>
+        `;
+      })
+    }
   },
 
-  displayTotalBill(num) {
+  populateTotalBill() {
+    const total = hotel.currentCustomer.calculateTotalSpent(hotel.rooms);
     billSection.innerHTML = '';
     billSection.innerHTML += `
-      <p class="total-spent">$${num}</p>
+      <p class="total-spent">$${total}</p>
     `;
   },
 
-  displayDashboard(hotel, bookings, num) {
-    domUpdates.displayBookings(hotel, bookings);
-    domUpdates.displayTotalBill(num);
+  hideLoginPage() {
+    domUpdates.hide([loginPage]);
     main.classList.remove('align-left');
     main.classList.add('align-center');
-    domUpdates.show([customerDashboard]);
-    domUpdates.hide([loginPage]);
-    console.log(hotel);
+  },
+
+  displayCustomerDashboard() {
+    domUpdates.show([customerDashboard, menu]);
+    domUpdates.hide([bookingDashboard]);
+    dashboardMenuButton.classList.add('clicked');
+    bookMenuButton.classList.remove('clicked');
+    domUpdates.populateBookings();
+  },
+
+  displayBookingDashboard() {
+    domUpdates.show([bookingDashboard]);
+    domUpdates.hide([customerDashboard]);
+    dashboardMenuButton.classList.remove('clicked');
+    bookMenuButton.classList.add('clicked');
+    domUpdates.setMinDate();
+    domUpdates.populateAvailableRooms();
+  },
+
+  checkFilters() {
+    const types = [];
+    typeFilters.forEach(checkbox => {
+      if (checkbox.checked) {
+        types.push(checkbox.value.replace('-', ' '));
+       }
+    });
+    if (types.length > 0) {
+      hotel.filterRooms(types);
+    }
+  },
+
+  bookRoom(e) {
+    const date = dateInput.value.replaceAll('-', '/')
+    const roomNumber = parseInt(e.target.id)
+    hotel.currentCustomer.createNewBooking(date, roomNumber)
```

Comment:
Wanted to get a second pair of eyes on this nested then statement. It seems to be working as expected! Any thoughts or edits you notice for this? 

How it works:
createNewBooking is a Customer method that calls on my api post function. So, createNewBooking returns the promise from the post. The first then statement waits for the post to be completed, then calls the api function to get the updated bookings api - fetchBookings() returns that promise and the next then statement uses the bookings data to update the allBookings property of my Hotel instance and the myBookings property of the current customer

Reply:
You know as well as I do how these chained .**then(=>** statements work. If everything seems to be instantiating correctly then go for it. I am under the impression is you'll want a catch at the end of all **.then(data =>** statements that actually processes the date that is returned.

Since this is nested, the **.catch(error =>** shouldn't be "chained" so I don't believe they will conflict with each other